aspnetcore/aspnetcore-2.1.md

```diff
@@ -0,0 +1,221 @@
+* Setting the content root the application's project root so that static files and views can be found.
+* Setting up the application on TestServer
+
+An implementation of the new test fixture with [xUnit](https://xunit.github.io/) would look like this
+
+```csharp
+using Xunit;
+namespace MyApplication.FunctionalTests
+{
+    public class MyApplicationFunctionalTests : IClassFixture<WebApplicationTestFixture<Startup>>
+    {
+        public MyApplicationFunctionalTests(WebApplicationTestFixture<Startup> fixture)
+        {
+            Client = fixture.Client;
+        }
+
+        public HttpClient Client { get; }
+
+        [Fact]
+        public async Task GetHomePage()
+        {
+            // Arrange & Act
+            var response = await Client.GetAsync("/");
+            // Assert
+            Assert.Equal(HttpStatusCodes.OK, response.StatusCode);
+        }
+    }
+}
+```
+
+For more information, see the [Announcement on GitHub](https://github.com/aspnet/announcements/issues/275)
+
+## [ApiController], ActionResult
+
+In ASP.NET Core 2.1, improved support for OpenAPI specification(previously known as **Swaggger**)) has been added to Web API. `ActionResult<T>` is a new type added to allow an application to return either a response type or any other action result (similar to IActionResult), while still indicating the response type. The `[ApiController] attribute has also been added as the way to opt-in to Web API specific conventions and behaviors.
+
+For more information, see [Build wbe APIs with ASP.NET Core](xref:web-api)
+
+## IHttpClientFactory
+
+The new `HttpClientFactory` type can be registered and used to configure and consume instances of HttpClient in applications. HttpClient already has the concept of delegating handlers that could be linked together for outgoing HTTP requests. The factory will make registering of these per named client more intuitive as well as implement a Polly handler that allows Polly policies to be used for Retry, CircuitBreakers, etc.
+
+For more information, see [Initiate HTTP Requests](xref:fundamentals/http-requests)
+
+## Kestrel on Sockets
+
+In ASP.NET Core 2.1, a new socket transport has been added to Kestrel. An application can be configured to use this new transport by adding the `.UseSockets()` extension to the `HostBuilder`
+
+```csharp
+public static IWebHost BuildWebHost(string[] args) =>
+    WebHost.CreateDefaultBuilder(args)
+    .UseSockets()
+    .UseStartup<Startup>()
+ ```
+
+## Generic host builder
+
+In ASP.NET Core 2.1, the Generic Host Builder (`HostBuilder`) has been introduced. This Builder can be used for applications that do not process HTTP requests (Messaging, background tasks, etc).
+
+For more information, see [.NET Generic Host](xref:fundamentals/host/generic)
+
+## Updated SPA templates
+
+In ASP.NET Core 2.1, the Single Page Application Templates for Angular, React, and React with Redux have been updated to account for changes in the architecture of each framework. 
```

Comment:
@SteveSandersonMS Is "architecture" the right word here? Do we have a writeup somewhere of the changes that were done to the SPA templates?

Reply:
I'd say something more like:

> In ASP.NET Core 2.1, the Single Page Application templates for Angular, React, and React with Redux have been updated to use the standard project structures and build systems for each framework. The Angular template is now based on Angular CLI, and the React templates are now based on create-react-app.

We don't have a writeup on the changes, because it's hard to say anything useful other than what's in the block above. It's not like there's some delta between old and new where we can say "X is now Y" other than "the templates are now different".

Reply:
How about "CLI friendly"?

Reply:
:+1: